datasource/read.go

```diff
@@ -22,6 +31,10 @@ type ReadRequest struct {
 
 	// ProviderMeta is metadata from the provider_meta block of the module.
 	ProviderMeta tfsdk.Config
+
+	// ClientCapabilities defines optionally supported protocol features for the
+	// ReadDataSource RPC, such as forward-compatible Terraform behavior changes.
+	ClientCapabilities *ReadClientCapabilities
```

Comment:
(I'm okay with simplifying the _request_ side of this for consumers if we want -- we had discussed out of band the implications of pointers for the response side and a preference for explicitness, but for this particular situation, there's not a semantic difference whether or not the client capabilities were actually announced since all fields should be disabled by default in the protocol when they are added.)